extern/sector-storage/fsutil/filesize_unix.go

```diff
@@ -11,19 +12,29 @@ type SizeInfo struct {
 	OnDisk int64
 }
 
-// FileSize returns bytes used by a file on disk
+// FileSize returns bytes used by a file or directory on disk
+// NOTE: We care about the allocated bytes, not file or directory size
 func FileSize(path string) (SizeInfo, error) {
-	var stat syscall.Stat_t
-	if err := syscall.Stat(path, &stat); err != nil {
-		if err == syscall.ENOENT {
+	var size int64
+	err := filepath.Walk(path, func(_ string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if !info.IsDir() {
+			stat := info.Sys().(*syscall.Stat_t)
```

Comment:
Given that this is a `_unix.go` file, I think we must always have a value in `Sys()`, so I don't think we need to check for nil pointer here.